sn_node/src/node/messaging/mod.rs

```diff
@@ -78,10 +81,11 @@ impl MyNode {
 
         match msg_type {
             MsgType::Node { msg_id, dst, msg } => {
+                let read_locked_node = node.read().await;
                 // Check for entropy before we proceed further
                 // Anythign returned here means there's an issue and we should
                 // short-circuit below
-                let ae_cmds = self
+                let ae_cmds = read_locked_node
```

Comment:
```suggestion
                let ae_cmds = node.read().await
```

This doesn't make a difference currently, but in the future it might prevent keeping the lock just that little bit longer than needed.